src/System.IO.Ports/tests/Legacy/SerialPort/SerialPortRegressions.cs

```diff
@@ -0,0 +1,138 @@
+    private int _numTestcases = 0;
+    private int _exitValue = TCSupport.PassExitCode;
+    private static string s_receivedstr = "";
+    private static SerialPort s_com1;
+    private static SerialPort s_com2;
+    private static bool s_readComplete = false;
+
+
+    public static void Main(string[] args)
+    {
+        SerialPortRegressions objTest = new SerialPortRegressions();
+        AppDomain.CurrentDomain.UnhandledException += new UnhandledExceptionEventHandler(objTest.AppDomainUnhandledException_EventHandler);
+
+        Console.WriteLine(s_strTFPath + " " + s_strTFName + " , for " + s_strClassMethod + " , Source ver : " + s_strDtTmVer);
+
+        try
+        {
+            objTest.RunTest();
+        }
+        catch (Exception e)
+        {
+            Console.WriteLine(s_strTFAbbrev + " : FAIL The following exception was thorwn in RunTest(): \n" + e.ToString());
+            objTest._numErrors++;
+            objTest._exitValue = TCSupport.FailExitCode;
+        }
+
+        ////	Finish Diagnostics
+        if (objTest._numErrors == 0)
+        {
+            Console.WriteLine("PASS.	 " + s_strTFPath + " " + s_strTFName + " ,numTestcases==" + objTest._numTestcases);
+        }
+        else
+        {
+            Console.WriteLine("FAIL!	 " + s_strTFPath + " " + s_strTFName + " ,numErrors==" + objTest._numErrors);
+
+            if (TCSupport.PassExitCode == objTest._exitValue)
+                objTest._exitValue = TCSupport.FailExitCode;
+        }
+
+        Environment.ExitCode = objTest._exitValue;
+    }
+
+    private void AppDomainUnhandledException_EventHandler(Object sender, UnhandledExceptionEventArgs e)
+    {
+        _numErrors++;
+        Console.WriteLine("\nAn unhandled exception was thrown and not caught in the app domain: \n{0}", e.ExceptionObject);
+        Console.WriteLine("Test FAILED!!!\n");
+
+        Environment.ExitCode = 101;
+    }
+
+
+    //This test is a regression test for DevDivBugs 14181: SerialPort: Data corruption occurs if 
+    //DataReceived event is used to receive Unicode characters sent across serial ports
+    public bool RunTest()
+    {
+        bool retValue = true;
+        TCSupport tcSupport = new TCSupport();
+
+        retValue &= tcSupport.BeginTestcase(new TestDelegate(UTF8Encoding), TCSupport.SerialPortRequirements.LoopbackOrNullModem);
+
+        _numErrors += tcSupport.NumErrors;
+        _numTestcases = tcSupport.NumTestcases;
+        _exitValue = tcSupport.ExitValue;
+
+        return retValue;
+    }
+
+    public bool UTF8Encoding()
+    {
+        if (!VerifyReadExisting(new System.Text.UTF8Encoding()))
+        {
+            Console.WriteLine("Err_018!!! Verifying readexisting");
+            return false;
+        }
+
+        return true;
+    }
+
+
+    public bool VerifyReadExisting(System.Text.Encoding encoding)
+    {
+        string text = "????????????4??????????????????,?11????????????????????????????????????????????????,????????????,??????????";
```

Comment:
Is this string something in unicode of some kind that hasn't come across right, or is that just GitHub?